src/librustdoc/passes/html_tags.rs

```diff
@@ -48,8 +48,10 @@ fn drop_tag(
     if let Some(pos) = tags.iter().rev().position(|(t, _)| *t == tag_name) {
         // Because this is from a `rev` iterator, the position is reversed as well!
         let pos = tags.len() - 1 - pos;
+        // If the tag is nested inside a "<script>", not warning should be emitted.
+        let should_not_warn = tags.iter().take(pos + 1).any(|(at, _)| at == "script");
```

Comment:
Sorry - I was using "script" as a shorthand for "any tag that introduces a syntax that's not HTML". So that includes at least `style`. @Manishearth might have more suggestions.
```suggestion
        // If the tag is nested inside a "<script>", not warning should be emitted.
        let should_not_warn = tags.iter().take(pos + 1).any(|(at, _)| at == "script" || at == "style");
```